hv.c

```diff
@@ -353,6 +352,7 @@ Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32,
     return hv_common(hv, NULL, key, klen, flags, action, val, hash);
 }
 
+/* flags might have HVhek_NOTSHARED set. If so, we need to ignore that. */
```

Comment:
Agree.

But I'm not sure what to do better. This comment documents a "trip hazard", so I think it needs to stay **somewhere**, but I'm not sure where/how. I seem to be suffering from writers block currently (took a long time to figure out the seemingly simple changes to perldelta), so I thought it better to push the rebase with what I had, rather than stall further.

Reply:
It might make more sense either directly by `flags &= ~HVhek_NOTSHARED`, or above the `if (keysv)` block that includes it (in which case it probably needs to clarify that this chunk's job is to sanitize `flags`).

Reply:
> It might make more sense either directly by `flags &= ~HVhek_NOTSHARED`, or above the `if (keysv)` block that includes it (in which case it probably needs to clarify that this chunk's job is to sanitize `flags`).

I did this, expanding it with the clarification. It works much better there. Thanks for figuring out this improvement.